pkg/k8shandler/deployment.go

```diff
@@ -472,6 +456,46 @@ func (node *deploymentNode) update(upgradeStatus *api.ElasticsearchNodeStatus) e
 	return nil
 }
 
+func (node *deploymentNode) executeUpdate() error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// isChanged() will get the latest revision from the apiserver
+		// and return false if there is nothing to change and will update the node object if required
+		if node.isChanged() {
+			if updateErr := sdk.Update(&node.self); updateErr != nil {
+				logrus.Debugf("Failed to update node resource %v: %v", node.self.Name, updateErr)
+				return updateErr
+			}
+		}
+		return nil
+	})
+}
+
+func (node *deploymentNode) updateStuckPod(upgradeStatus *api.ElasticsearchNodeStatus) error {
+	if node.isChanged() {
+		logrus.Infof("Requested to update node '%s', which is unschedulable. Skipping rolling restart scenario and performing redeploy now", upgradeStatus.DeploymentName)
+
+		node.unpause()
```

Comment:
i think the order here should be update the deployment, then unpause, wait for the roll out, then pause again. this keeps changes a bit more in lock-step.

Reply:
:+1: